ipi/pes/metatensor.py

```diff
@@ -271,13 +277,11 @@ def __call__(self, cell, pos):
                 # torch.autograd.functional.jacobian, which is vectorized
                 def _compute_ensemble(positions, strain):
                     new_system = mta.System(
-                        types, positions @ strain, cell @ strain, pbc
+                        self._types, positions @ strain, cell @ strain, pbc
                     )
                     for options in self.model.requested_neighbor_lists():
-                        # we meed to recompute the neighbor list to be able to register gradients
-                        neighbors = mta.ase_calculator._compute_ase_neighbors(
-                            ase_atoms, options, dtype=self._dtype, device=self.device
-                        )
+                        # get the pre-computed NL, and re-attach it in the compute graph
+                        neighbors = system.get_neighbor_list(options).detach()
```

Comment:
@frostedoyster do you have an example where `force_virial_ensemble` is used? I did not check that this code still works

Reply:
My bad, this would be `input-ensemble.xml`